controllers/topolvm_snapshotclass.go

```diff
@@ -72,10 +75,16 @@ func (s topolvmVolumeSnapshotClass) ensureDeleted(r *LVMClusterReconciler, ctx c
 
 		if err != nil {
 			// already deleted in previous reconcile
-			if errors.IsNotFound(err) {
+			if k8serrors.IsNotFound(err) {
 				r.Log.Info("topolvm volume snapshot class is deleted", "VolumeSnapshotClass", vscName)
 				return nil
 			}
+			if runtime.IsNotRegisteredError(err) || meta.IsNoMatchError(err) ||
```

Comment:
Yes you are completely right. This didnt get catched because a failure will actually not change the LVMCluster to failed it will just log "failed to reconcile". This has to be reflected properly in the status, but that will be tackled with https://issues.redhat.com/browse/OCPVE-627 when we revisit the resource creation / status checks

Reply:
I anyhow added the check now also to create obviously